src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -199,6 +203,13 @@ public IClient build() {
 
 			OkHttpClient.Builder builder = new OkHttpClient.Builder()
 				.addInterceptor(responseCodeInterceptor)
+				.addNetworkInterceptor(new Interceptor() {
+                    @Override
+                    public Response intercept(Chain chain) throws IOException {
+                        Request agent = chain.request().newBuilder().header("User-Agent", "openshift-restclient-java/"+Version.userAgent()).build();
```

Comment:
Might it be more useful to allow a user of the ClientBuilder to set the useragent so that you could determine from which application requests are originating?  Isn't that part of the problem we are trying to address here, who is the bad actor?